CMakeLists.txt

```diff
@@ -1,11 +1,12 @@
-cmake_minimum_required(VERSION 3.18 FATAL_ERROR)
+cmake_minimum_required(VERSION 3.25 FATAL_ERROR)
 # cmake_policy(SET CMP0022 NEW) cmake_policy(SET CMP0023 NEW)
 
 # Use compiler ID "AppleClang" instead of "Clang" for XCode. Not setting this
 # sometimes makes XCode C compiler gets detected as "Clang", even when the C++
 # one is detected as "AppleClang".
 cmake_policy(SET CMP0010 NEW)
 cmake_policy(SET CMP0025 NEW)
+cmake_policy(SET CMP0126 OLD)
```

Comment:
Whelp, when we loose support here for `<3.25` I assume this CMake policy will break @cyyever . Therefore, we need to properly fix this and update the CMake scripts to NEW to upgrade past 3.21

Reply:
We can't set it to `NEW` because that may break third-party libraries, they are out of our control.

Reply:
Yeah, but CMake will also break when we update to a newer version and this policy OLD is no longer supported. do you have any idea which libraries will be problematically in third_party? Or is it just libraries which include PyTorch? Seems like libkineto is definitely one, but we have control over that one.

Reply:
Let's remove the line and see what build issues here.

Reply:
It has to be set to old because the options of some third-party libraries are set by cache and it's not easy to fix them.